### PR TITLE
Updating dp-kafka

### PR DESCRIPTION
**Pull request by franmoore05**
### What

Updated dp-kafka library to ensure correct healthcheck status is returned if a broker is not available.
Update go version

### How to review

Ensure the tests pass and the changes make sense.

### Who can review

Anyone.

### CODE REVIEW

File: cmd/dp-import-reporter/main.go [hunk 1]
```diff
@@ -51,7 +51,8 @@ func main() {
 	datasetAPIClient, err := client.NewDatasetAPIClient(cfg.ServiceAuthToken, cfg.DatasetAPIURL, cfg.DatasetAPIAuthToken, &http.Client{}, ResponseBodyReader{})
 	if err != nil {
 		log.Error(ctx, "error creating new dataset api client", err)
-		os.Exit(1)
+		//os.Exit(1)
+		return
 	}
```
Comment (s-xton) [inline cmd/dp-import-reporter/main.go]: correct ?

File: cmd/dp-import-reporter/main.go [hunk 2]
```diff
@@ -99,7 +100,8 @@ func main() {
 	)
 	if err != nil {
 		log.Error(ctx, "error while attempting to create kafka kafkaConsumer", err)
-		os.Exit(1)
+		//os.Exit(1)
+		return
 	}
```
Comment (s-xton) [inline cmd/dp-import-reporter/main.go]: as above

File: go.mod [hunk 3]
```diff
@@ -11,27 +11,27 @@ replace github.com/spf13/cobra => github.com/spf13/cobra v1.4.0
 
 require (
 	github.com/ONSdigital/dp-api-clients-go v1.43.0 // indirect
-	github.com/ONSdigital/dp-healthcheck v1.5.0 // indirect
-	github.com/ONSdigital/dp-kafka/v2 v2.7.3
+	github.com/ONSdigital/dp-healthcheck v1.6.1 // indirect
+	github.com/ONSdigital/dp-kafka/v2 v2.8.0
 	github.com/ONSdigital/dp-net v1.5.0
 	github.com/ONSdigital/go-ns v0.0.0-20210916104633-ac1c1c52327e
```
Comment (s-xton) [inline go.mod:17]: the indirect and actual seem to be mixed in ?

File: go.mod [hunk 4]
```diff
@@ -11,27 +11,27 @@ replace github.com/spf13/cobra => github.com/spf13/cobra v1.4.0
```
Comment (s-xton) [inline go.mod:11]: could we upgrade dp-api-client-go and dp-net to v2 ? Perhaps out of scope

**Review by s-xton — CHANGES_REQUESTED** [review]

File: cmd/dp-import-reporter/main.go (hunk 1 above)
Comment (franmoore05) [reply cmd/dp-import-reporter/main.go]: This has now been removed

File: go.mod (hunk 3 above)
Comment (franmoore05) [reply go.mod:17]: This is how I picked it up from the main branch so I don't want to change it.

File: go.mod (hunk 4 above)
Comment (franmoore05) [reply go.mod:11]: Out of scope for this urgent MSK fix.

**Review by s-xton — APPROVED** [review]

**Pull request merged**